app/assets/scripts/vendor/map.js

```diff
@@ -74,4 +86,56 @@
         setupClickListener('changetype-all', []);
         setupClickListener('changetype-address', ['address']);
         setupClickListener('changetype-establishment', ['establishment']);
+
+        var mapViewButton = document.getElementById("map_view");
+        var listViewButton = document.getElementById("list_view");
+
+        mapViewButton.addEventListener('click', function() {
+            $("#map").show();
+            $("#list").hide();
+        });
+
+        listViewButton.addEventListener('click', function() {
+            $("#map").hide();
+            populateListDiv();
+            $("#list").show();
+        });
+
       }
+
+      function populateListDiv() {
+          var markersToShow = [];
+          $("#list").empty();
+          var bounds = map.getBounds();
+          for (var i = 0; i < markers.length; i++) {
+              if (bounds.contains(markers[i].getPosition())) {
+                  markersToShow.push(markers[i]);
+              }
+          }
+          var table = $("<table border=1></table>");
+          $.each(markersToShow, function(i, markerToShow) {
+            tableCell = document.createElement('td');
+            $(tableCell).attr({
+              'style': 'overflow:hidden;width:100%;height:100%;position:absolute'
+            });
+            tableCellBoldTitle = document.createElement('strong');
+            $(tableCellBoldTitle).html(markerToShow.getTitle());
+            $(tableCell).append(tableCellBoldTitle);
+            tableCellNewline = document.createElement('br');
+            $(tableCell).append(tableCellNewline);
+            $(tableCell).append(markerToShow.getLabel());
```

Comment:
nit: Can you group the three appends, such as ` $(tableCell).append(tableCellBoldTitle, tableCellNewline, markerToShow.getLabel());`